2022-07-20
Attendees:
- Nate Olson: at NIST for 10 years, Perl in masters, now in R, got PhD in bioinformatics from Maryland, figure generation for papers, data analysis, ggplot2, project management for the team
- Sierra Miller: started in physics, but switched to degree in molecular bio to work on genomics, 2 years in wet lab, then switched to bioinformatics to analyze RNAseq data, onboarding wet lab and analysis pipelines, off target assays (unintended edits during CRISPR), sources of variability, does depth increase off-target detection?
  - output format is typically HTML from R Markdown
- Douglas Boldt: 18 months at aws, telehouse america, lots of sys admin, automation, linux, networkings, CLI, API, SEO
- Dan Shepard: development rep, previously at deltek working on government contracting software (govwin), previously consulting, mostly in disaster management (getting money from FEMA), CARES Act
- Matt Dowling: account rep, 2 years at AWS, works alongside Derrick, worked in Dan's role previously, previously at Oracle for a year
- Adei: medical device innovation consortium, using giab cell line samples for CRISPR genome editing, mixtures of edited cells, validate variant detection techniques to detect variants created with CRISPR, mixture study, validate with qPCR

- snakemake: run using lambda to spin up ec2 and store data in s3, redshift
  - https://snakemake.github.io/
- tibanna:
  - https://tibanna.readthedocs.io
  - cost tracking
- aws genomics cli: https://aws.amazon.com/genomics-cli
- problems with nist firewall while working with tibanna
  - default dns for docker containers is blocked by nist firewall
  - pre command to modify etc host file to change the resolver
  - can request to open up firewall

* Data management and analysis platform, receive data from external collaborators
  - Data Portal/Registry needs to be:
    - queryable
    - API
    - Dashboard for ...
Want data warehouse
make it easy to download data from data lake
- Inputs:
  - genome sequencing data,
    - RNAseq,
    - info on
      - concentration,
      - fragment size,
    - collaborator contributed sequencing,
    - primary analysis:
      - read alignments,
      - variant calls,
      - flat and binary files,
      - bgzipped files
- FTP site hosted by NIH, 8-10 years old, there's a NIH collaborator who maintains a github site with information, use AWS Transfer, metadata cataloging is slow manual process, documented
- Human Pan-genome Research Consortium (HPRC) hosts public S3 buckets, have lots of sequencing data, vcf, bam, binary assemblies, fasta/fastq
- team at Google Health has done some sequencing
- NCBI SRA has the best metadata
- S3 and google drive for data that is not ready to be made public
- Sagemaker immersion day
- in-person training: 
- group access control is very difficult for customer
- globus has nice group access control, nice connection to s3
- full nist data structure in aws
- terabytes of data in aws
- nist public data: data.nist.gov (talk to Erin Chu)
  - download cart
- focus on nate's requirements
- nate has s3 connectors set up
- publications depend on links to sftp site

EDA, bioinfo pipeline dev (snakemake)
- provenance tracking via rmarkdown and jupyter notebooks
  - using shournal: https://github.com/tycho-kirchner/shournal
- galaxy
- EDA: mostly rstudio and jupyter notebooks, VSCode
- need for logging
- documentation and reproducibility is hard to do, needs to be automated
- predefined workflows, e.g. qc workflows
- create doi for provenance and reproducibility
- application development side: execute queries, nlp, elasticsearch
- working EKS is difficult
- FAIR principles
- open source:
- Redshift for data warehouse
  - create pointers
  - federation
  - avoid data duplication
  - tableau
  - search thru the data warehouse (query)
- lay of the land in terms of what data are available
- eda: quality control, understanding genomes, basic stats
- generate bench marks sets, characterization
- reference genomes are 7 individuals, nist has a lot of data on these individuals
- collaborators do assembly
- explainable boosting machines (ebm)
- issues running snakemake on aws, necessitates using shared workstations / clusters maintained by different groups at nist
- ask justin about where data is, he has a good memory
- nist researchers and collaborators don't know where data is
- documented and transparent
- reporting
- runs pipeline, shell commands, snakemake, sync with google drive
- example of dashboard: https://www.internationalgenome.org/data-portal/sample
- nih ncbi trace ftp
- ftp site data is mirrored on aws s3
- hundreds of terabytes, ftp site 20-40 terabytes
- cell culture from blood
- DNA extraction
- aliquot into vials
- sequencing
  - short read
    - illumina
  - long read
    - pacbio
- read mapping, align to reference genome
- use case
- benchmarks and characterization
- example: 23andMe and ancestry.com
- variants for medical diagnoses
- NCBI sequence read archive (SRA)
- sequencing technologies:
  - 10x genomics (short reads with barcodes)
  - pacbio (long read)
  - illumina (short read)
  - nanopore (long read)
- Is redshift ATO'd? Only about two dozen services are enabled for all of NIST
- CloudFormation templates will include cost allocation tags
- Quality report
- read length distribution
- coverage for bam files
- 3 units
